dix/req.c

```diff
@@ -1625,13 +1625,32 @@ static void dix_rop_completed(struct m0_sm_group *grp, struct m0_sm_ast *ast)
 	struct m0_dix_rop_ctx *rop_del_phase2 = NULL;
 	bool                   del_phase2 = false;
 	struct m0_dix_cas_rop *cas_rop;
+	uint32_t	       req_cnt = 0;
+	int		       rc;
+	bool		       cas_success = false;
 
 	(void)grp;
 	if (req->dr_type == DIX_NEXT)
 		m0_dix_next_result_prepare(req);
 	else
-		m0_tl_for(cas_rop, &rop->dg_cas_reqs, cas_rop) {
-			dix_cas_rop_rc_update(cas_rop, 0);
+		/*
+		 * Check return status of N cas request, if any one CAS
+		 * request is success, we return success to client. For failed
+		 * request skipping to send rc_update() but if it is last request
+		 * and we do not found any success request reply so far then
+		 * do update rc_update() for last failed CAS request also.
```

Comment:
The tail of the dg_cas_reqs list is not the last CAS request. Please re-phase this sentence in a way that it matches with the code below.

Reply:
Is this comment with respect to the code changes you proposed ?  If yes then I agree that this need to rephrase if Opted for the changes you proposed.